code/scripting/api/objs/LuaEnum.cpp

```diff
@@ -77,5 +77,32 @@ ADE_FUNC(addEnumItem,
 	return ADE_RETURN_TRUE;
 }
 
+ADE_FUNC(removeEnumItem,
+	l_LuaEnum,
+	"string itemname",
+	"Removes an enum item with the given string.",
+	"boolean",
+	"Returns true if successful, false otherwise")
+{
+	lua_enum_h lua_enum;
+	const char* item_name;
+	if (!ade_get_args(L, "os", l_LuaEnum.Get(&lua_enum), &item_name))
+		return ADE_RETURN_FALSE;
+
+	size_t i;
+
+	for (i = 0; i < lua_enum.getEnum()->list.size(); i++) {
+		if (stricmp(item_name, lua_enum.getEnum()->list[i].c_str())) {
+			return ADE_RETURN_FALSE;
+		} else {
```

Comment:
I have to be reading this wrong.  If any item on the list is different than the provided name, abort the function and return false?

Reply:
See, this is why I shouldn't test using only the first item in the list. Good catch. Fixed.